### PR TITLE
Use TestingTB interface instead of testing.TB

### PR DESCRIPTION
**Pull request by royvandewater**
Use `TestingTB` interface instead of `testing.TB`

`TestingTB` allows for non `*testing.T` instances to be passed in. This allows the testing utils to work with other test runners like Ginkgo.

### CODE REVIEW

**Comment by royvandewater** [comment]
Not sure why the tests are marked as failed, GH seems to not be getting status updates from CCI, but the tests passed here: https://app.circleci.com/pipelines/github/circleci/ex/560/workflows/024d016a-c6d8-4250-a104-1ff463ea0a12

File: testing/types/doc.go
Comment (marcomorain) [inline testing/types/doc.go:4]: As a newcomer to Go, I'm curious why this is pulled out into a separate file?

Comment (royvandewater) [reply testing/types/doc.go:4]: I saw it being done in other packages in this repo, so I copied that pattern. I think the idea is to make an consistent place to look for the package docstring.

Comment (MaT1g3R) [reply testing/types/doc.go:4]: It's a convention for package level documentation (https://go.dev/blog/godoc)

Comment (SemanticallyNull) [reply testing/types/doc.go:4]: In general types like this are put into their own package so that you don't end up with circular dependencies inside packages.

**Comment by MaT1g3R** [comment]
`testing.TB` is already an interface, so I don't know why this is needed

**Comment by MaT1g3R** [comment]
I think generally my opinion on this is to use the standard library for testing as much as possible. I usually can't justify bringing in another dependency for tests when the standard library implementation already works well.

That being said if there's a very strong preference for Ginkgo I won't block this, but I hope that teams can at least give the standard library a go before switching out for something else.

I will leave it up to @pete-woods to review this, as I feel he might have something to say as well.

File: testing/types/doc.go
Comment (pete-woods) [reply testing/types/doc.go:4]: More typical would be to have an internal type where it's used, like gotest.tools does:

https://github.com/gotestyourself/gotest.tools/blob/main/assert/assert.go#L77

I don't strongly have an issue with putting it in a common package, but it should be in an `/internal/types` subpackage.

**Comment by pete-woods** [comment]
> I think generally my opinion on this is to use the standard library for testing as much as possible. I usually can't justify bringing in another dependency for tests when the standard library implementation already works well.
> 
> That being said if there's a very strong preference for Ginkgo I won't block this, but I hope that teams can at least give the standard library a go before switching out for something else.
> 
> I will leave it up to @pete-woods to review this, as I feel he might have something to say as well.

I do agree, but if people really want to use ginkgo I won't stop them, as long as it doesn't excessively complicate our code here. I'm definitely not in the business of trying to dictate technology choices to people.

File: testing/types/doc.go
Comment (pete-woods) [reply testing/types/doc.go:4]: > As a newcomer to Go, I'm curious why this is pulled out into a separate file?

I think you only do this if you've got more than one file in the package. For a single-file package like this, I'd tend to do it inline.

Having it in a dedicated `doc.go` file makes it much easier to discover for those not using IDEs.

**Review by MaT1g3R — APPROVED** [review]

File: testing/types/doc.go
Comment (pete-woods) [reply testing/types/doc.go:4]: https://github.com/circleci/ex/blob/main/testing/httprecorder/ginrecorder/middleware.go

**Comment by royvandewater** [comment]
Is there anything extra we need to do to bump the package version before merging?

**Comment by MaT1g3R** [comment]
No, it's good to go

**Comment by MaT1g3R** [comment]
FWIW: versioning is automatic https://github.com/circleci/ex/blob/0cc65a6d3671323f7c010186179fc895878cb80b/do#L6

**Pull request merged**
